src/terminal/writer.go

```diff
@@ -392,17 +400,18 @@ func write(s rune) {
 		return
 	}
 
-	if !Interactive {
-		for special, escape := range formats.EscapeSequences {
-			if s == special && lastRune != escape {
-				builder.WriteRune(escape)
-			}
+	// UNSOLVABLE: When "Interactive" is true, the prompt length calculation in Bash/Zsh can be wrong, since the final string expansion is done by shells.
+	length += runewidth.RuneWidth(s)
+	// length += utf8.RuneCountInString(string(s))
+
+	if !Interactive && !Plain {
```

Comment:
A problem here: when `Interactive` is `true`, a series of backslashes will be half eaten by the shell escape logic.